homeassistant/components/denonavr/media_player.py

```diff
@@ -524,3 +569,58 @@ async def async_set_dynamic_eq(self, dynamic_eq: bool):
 
         if self._update_audyssey:
             await self._receiver.async_update_audyssey()
+
+    @async_log_errors
+    async def async_send_telnet_commands(self, commands: str) -> list[dict[str, str]]:
+        """Send telnet commands to receiver and subscribe to its responses."""
+        command_list = commands.splitlines()
+        events = []
+        for command in command_list:
+            found = False
+            for event in TELNET_EVENTS_SORTED:
+                if command.startswith(event):
+                    events.append(event)
+                    found = True
+                    break
+            if not found:
+                raise HomeAssistantError(
```

Comment:
https://developers.home-assistant.io/docs/integration_quality_scale_index?_highlight=valueerror#silver-

```suggestion
                raise ValueError(
```